### PR TITLE
feat(gatsby-plugin-gatsby-cloud): Emit Routes via IPC

### PR DESCRIPTION
**Pull request by abhiaiyer91**
Emit relevant routes via IPC

### CODE REVIEW

File: packages/gatsby-plugin-gatsby-cloud/src/gatsby-node.js [hunk 1]
```diff
@@ -29,14 +29,41 @@ exports.onCreateWebpackConfig = ({ actions, stage }) => {
   })
 }
 
+function emitRoutes(routes) {
+  if (!process.send) {
+    console.log(`No process send available`)
```
Comment (wardpeet) [inline packages/gatsby-plugin-gatsby-cloud/src/gatsby-node.js:34]: Should we log this?

**Review by wardpeet — DISMISSED** [review]
Looks good, some small unit tests would be great :)

File: packages/gatsby-plugin-gatsby-cloud/src/gatsby-node.js (hunk 1 above)
Comment (abhiaiyer91) [reply packages/gatsby-plugin-gatsby-cloud/src/gatsby-node.js:34]: No will remove

**Comment by abhiaiyer91** [comment]
Yes i'll add some tests

File: package.json [hunk 2]
```diff
@@ -152,6 +152,7 @@
     "test:peril": "cd peril && yarn test",
     "test:update": "jest --updateSnapshot",
     "test:watch": "jest --watch",
+    "test:specific": "jest",
```
Comment (wardpeet) [inline package.json:155]: Why this? You can also do `yarn jest` no need to have a specific command
```suggestion
```

File: packages/gatsby-plugin-gatsby-cloud/src/__tests__/routes.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      action: {
+        type: `LOG_ROUTES`,
+        payload: {
+          routes: {
+            "path/2/index.html": `SSR`,
+            "page-data/path/2/page-data.json": `SSR`,
+          },
+        },
+      },
+    })
```
Comment (wardpeet) [inline packages/gatsby-plugin-gatsby-cloud/src/__tests__/routes.js:81]: Should we add a not havebeen called with test?
```suggestion
    expect(process.send).toHaveBeenCalledWith({
      type: `LOG_ACTION`,
      action: {
        type: `LOG_ROUTES`,
        payload: {
          routes: {
            "path/2/index.html": `SSR`,
            "page-data/path/2/page-data.json": `SSR`,
          },
        },
      },
    })
    
    expect(process.send).not.toHaveBeenCalledWith({
      type: `LOG_ACTION`,
      action: {
        type: `LOG_ROUTES`,
        payload: {
          routes: {
            "path/3/index.html": `SSG`,
            "page-data/path/3/page-data.json": `SSG`,
          },
        },
      },
    })
```

**Review by wardpeet — COMMENTED** [review]
Added some small nits

File: packages/gatsby-plugin-gatsby-cloud/src/gatsby-node.js [hunk 4]
```diff
@@ -29,14 +29,41 @@ exports.onCreateWebpackConfig = ({ actions, stage }) => {
...
+    return
+  }
+
+  process.send({
+    type: `LOG_ACTION`,
```
[the remark this answers is not in the document]
Comment (abhiaiyer91) [reply packages/gatsby-plugin-gatsby-cloud/src/gatsby-node.js:39]: The only actions we support in cloud runner are `JobCreatedAction`, `IncBuildsAction`, and `LogAction`. And we use this `LogAction` for handling events

File: packages/gatsby-plugin-gatsby-cloud/src/gatsby-node.js [hunk 5]
```diff
@@ -29,14 +29,41 @@ exports.onCreateWebpackConfig = ({ actions, stage }) => {
...
   { store, pathPrefix, reporter },
   userPluginOptions
 ) => {
   const pluginData = makePluginData(store, assetsManifest, pathPrefix)
   const pluginOptions = { ...DEFAULT_OPTIONS, ...userPluginOptions }
 
-  const { redirects, pageDataStats, nodes } = store.getState()
+  const { redirects, pageDataStats, nodes, pages } = store.getState()
+
+  console.log(`Emitting Routes`)
```
[the remark this answers is not in the document]
Comment (abhiaiyer91) [reply packages/gatsby-plugin-gatsby-cloud/src/gatsby-node.js]: This log has been removed

File: package.json (hunk 2 above)
Comment (abhiaiyer91) [reply package.json:155]: this was a mistake

**Pull request closed without merging**